platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/OidcCredentialsResolver.java

```diff
@@ -47,50 +50,62 @@
 import org.pac4j.oidc.client.OidcClient;
 import org.pac4j.oidc.config.OidcConfiguration;
 import org.pac4j.oidc.credentials.OidcCredentials;
-import org.pac4j.oidc.credentials.authenticator.OidcAuthenticator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CustomOidcAuthenticator extends OidcAuthenticator {
+public class OidcCredentialsResolver {
```

Comment:
📘 This class was renamed to make more sense. Also, the token validation logic was moved entirely into this class, since it is the class that deals with the tokens.

Reply:
❗️ I still think it should extend the `OidcAuthenticator` to get the validation stuff the `OidcAuthenticator` does in the constructor.

Reply:
After looking in the `OidcAuthenticator`'s constructor, I think all it is doing is resolving the `ClientAuthentication` (i.e. Passing the `ClientId` and `Secret` to keycloak). We don't do that at the moment, but I think we should, and I think we could move that resolution logic into this class without having to extend

Reply:
We should probably do either one of those here for this PR

Reply:
- Extend `OidcAuthenticator`
- Remove `createClientAuthentication`